Assets/Scripts/Warrior.cs

```diff
@@ -131,12 +134,15 @@ private void Shooting()
     {
         _shootingTimer -= Time.deltaTime;
 
-        if (!IsDead && _isShooting && _shootingTimer <= 0)
+        if (_currentEnemy.UnitStats.Health >= 0)
```

Comment:
Create a property IsDead(bool) in UnitStats.
And return true if lives are zero or below
And check it when you check if the enemy is alive.

This approach hides the implementation of death in the UnitStats class. and thanks to this, the principle of encapsulation is realized. https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)

Reply:
Done